pyramid_openapi3/__init__.py

```diff
@@ -93,24 +93,20 @@ def openapi_validated(request: Request) -> dict:
         )
         # OpenAPI Core 0.16.0 has `validated.body` as a dataclass, however
         # 0.16.1 does not (and is just a dictionary)
+        # This is because 0.16.0 made dynamic dataclasses the default behaviour,
+        # whereas, 0.16.1 changed it back to an opt-in
+        # Issue: https://github.com/p1c2u/openapi-core/issues/430
+        # ===
+        # TODO: TO-Discuss
+        # We could remove this if statement and live with the differing behaviour
+        # between 0.16.0 and 0.16.1 as it is due to the openapi-core library.
+        # We could also just set the minimum supported version of this library to be
+        # 0.16.1 and sidestep the issue entirely
```

Comment:
If we set minimum version to 0.16.1, we still need this `if` because some people might opt into `dynamic dataclasses` feature of openapi-core, right?

Reply:
In that case we could leave out the `if` and let it return the `dynamic dataclasses` (as the user had to manually opt into it, and as such should expect that it would return as a `dataclass` and not a `dict`).

Right now this `if` would prohibit the user from opting into `dynamic dataclasses` as we would end up just converting it back into a `dict` here.

This `if` was added as a workaround due to `openapi-core` making `dynamic dataclasses` the default behaviour in 0.16.0 (and then reverting it in 0.16.1)

Reply:
Ahhh, that's nice! Let's pin to 0.16.1 as minimal version then!